front/vite.config.ts

```diff
@@ -1,41 +1,48 @@
 /// <reference types="vitest" />
-import { defineConfig } from 'vite';
+import { defineConfig, loadEnv } from 'vite';
 import react from '@vitejs/plugin-react-swc';
 import viteTsconfigPaths from 'vite-tsconfig-paths';
 import svgrPlugin from 'vite-plugin-svgr';
 import ImportMetaEnvPlugin from '@import-meta-env/unplugin';
 import checker from 'vite-plugin-checker';
 
 // https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [
-    react(),
-    viteTsconfigPaths(),
-    svgrPlugin(),
-    ImportMetaEnvPlugin.vite({
-      example: '.env.example',
-    }),
-    checker({
-      typescript: {
-        buildMode: true,
-      },
-      eslint: {
-        lintCommand: 'eslint --ext .ts,.tsx,.js,.jsx src --max-warnings 203',
+export default defineConfig(({ mode }) => {
+  const env = loadEnv(mode, process.cwd(), '');
+  return {
+    plugins: [
+      react(),
+      viteTsconfigPaths(),
+      svgrPlugin(),
+      ImportMetaEnvPlugin.vite({
+        example: '.env.example',
+      }),
+      checker({
+        typescript: {
+          buildMode: true,
+        },
+        eslint: {
+          lintCommand: 'eslint --ext .ts,.tsx,.js,.jsx src --max-warnings 203',
+        },
+        overlay: env.OSRD_VITE_OVERLAY !== 'false' && {
```

Comment:
```suggestion
        overlay: env.OSRD_VITE_OVERLAY === 'true' && {
```

Reply:
Yes and it's exactly what we want : if not assigned, it's true by default.